tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -129,3 +131,271 @@ def test_all_opcodes_in_container(
         data=eof_code,
         expect_exception=expect_exception,
     )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        op
+        for op in list(Op)
+        if op not in invalid_eof_opcodes
+        and op not in halting_opcodes
+        and op not in [Op.RJUMP, Op.RJUMPI, Op.RJUMPV]
+    ],
+)
+def test_all_invalid_terminating_opcodes(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all opcodes that are invalid as the last opcode in a container
+    """
+    if opcode.has_data_portion():
+        opcode_with_data_portion = opcode[0] if opcode != Op.CALLF else opcode[1]
```

Comment:
It took me 20 mins to figure out what it does.

Reply:
Added a comment here, and in some other confusing parts of the file for clarity.